app/decorators/controllers/solidus_multi_domain/taxons_controller_decorator.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module SolidusMultiDomain
+  module TaxonsControllerDecorator
+    def show
+      @taxon = ::Spree::Taxon.find_by_store_id_and_permalink!(current_store.id, params[:id])
```

Comment:
This change can be implemented on the `load_taxon` method to avoid overriding the `show` method.